Migrate ProjectionDocumentTest to JUnit5

vogella opened the pull request.

vogella committed:
Migrate ProjectionDocumentTest to JUnit5

vogella committed:
Migrate ProjectionMappingTest to JUnit5

HeikoKlare commented on tests/org.eclipse.text.tests/projection/org/eclipse/text/tests/ProjectionDocumentTest.java:
```diff
@@ -680,18 +679,18 @@ public void test9_7() {
 		} catch (BadLocationException e) {
-			assertTrue(false);
+			fail("BadLocationException occurred: " + e.getMessage());
 		}
```
nitpicky: the try-catch could be removed in favor of having the test method throwing the exception, such that more information than only the message is provided to the test executor.

HeikoKlare reviewed (commented):
The test class adaptations look good, thank you! Only a nitpicky comment (which I do not require to be changed in this PR).

Just note that you need to add the JUnit 5 dependencies to the MANIFEST and migrate the executing test suite (`ProjectionTestSuite`) to JUnit 5 as well.

vogella commented:
Thanks for the fast feedback @HeikoKlare, updated

vogella committed:
Migrate ProjectionTestSuite and EclipseTextTestSuite to JUnit5

vogella committed:
Add Jupiter API to org.eclipse.text.tests

HeikoKlare commented:
I just saw that the project contains two source folders and the project's "main" test suite (`EclipseTextTestSuite`) is placed in the one I missed before. That one needs to be migrated to JUnit 5 as well, as it is still based on JUnit 4 and can thus not execute the contained `ProjectionTestSuite` that has been migrated.

That's the reason why the bot currently reports 161 test cases (exactly the number of tests executed by `ProjectionTestSuite`:
```
This pull request removes 577 and adds 416 tests. Note that renamed tests count towards both.
```

The failing Jenkins build should hopefully be fixed in the next run. I have reported the issue (which happened to most PRs starting from yesterday afternoon on) to the helpdesk and they restarted the server: https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/4783

vogella committed:
Migrate ProjectionTestSuite and EclipseTextTestSuite to JUnit5

Instead of including the JUnit5 test suite in the EclipseTextTestSuite
test suite, include the tests. JUnit5 test suites seem to not support
included test suites.
